broker/src/main/java/io/zeebe/broker/system/partitions/impl/AsyncSnapshotDirector.java

```diff
@@ -199,7 +199,6 @@ private void persistSnapshotIfLastWrittenPositionCommitted() {
               if (pendingSnapshot != null
                   && lastWrittenEventPosition != null
                   && currentCommitPosition >= lastWrittenEventPosition) {
-
                 final var snapshotPersistFuture = pendingSnapshot.persist();
 
                 snapshotPersistFuture.onComplete(
```

Comment:
I like that persist is now idempotent :+1: Didn't expected that you remove the other change :) Isn;t it problematic when we call the onComplete listener twice? Since we reset the positions twice then? Happy to discuss this :)